### PR TITLE
adds docker-compose override file, adds tests to clarify tool behavior, updates libs version

### PR DESCRIPTION
**Pull request by ebaranetsky**

### CODE REVIEW

File: products/docker-compose.override.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+version: '3.4'
```
Comment (irublev) [inline products/docker-compose.override.yml:1]: Strange file name, what is overridden?

File: products/core/src/main/java/org/alliedium/ignite/migration/Controller.java [hunk 2]
```diff
@@ -66,6 +70,19 @@ public void serializeDataToAvro(Path rootSerializedDataPath) {
         tasksExecutor.shutdown();
     }
 
+    /**
+     * Represents a deserialize operation of avro file right into ignite.
```
Comment (irublev) [inline products/core/src/main/java/org/alliedium/ignite/migration/Controller.java]: ignite -> Ignite or even Apache Ignite

File: products/core/src/test/java/org/alliedium/ignite/migration/BadPathAsserter.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package org.alliedium.ignite.migration;
```
Comment (irublev) [inline products/core/src/test/java/org/alliedium/ignite/migration/BadPathAsserter.java]: Are you sure this class is necessary at all? There is a standard way to write negative tests in TestNG

File: products/core/src/test/java/org/alliedium/ignite/migration/SupportedActionsTests.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+        CacheConfiguration<Integer, PartialField> cacheConfiguration = new CacheConfiguration<>();
+        cacheConfiguration.setQueryEntities(Collections.singleton(queryEntity));
+        cacheConfiguration.setName(cacheName);
+
+        IgniteCache<Integer, PartialField> cache = ignite.createCache(cacheConfiguration);
+        List<PartialField> partialFields = new ArrayList<>();
+        for (int i = 0; i < 10; i++) {
+            partialFields.add(new PartialField("testName", "testType", "testVal"));
+            cache.put(i, partialFields.get(i));
```
Comment (irublev) [inline products/core/src/test/java/org/alliedium/ignite/migration/SupportedActionsTests.java]: Why do you need to store a list?

File: products/docker-compose.override.yml (hunk 1 above)
Comment (ebaranetsky) [reply products/docker-compose.override.yml:1]: https://docs.docker.com/compose/reference/#specifying-multiple-compose-files - official docker compose way to override container setting

Comment (irublev) [reply products/docker-compose.override.yml:1]: > also specifies this same service, **any matching** fields override the previous file

Thus, this is not the case, we have not defined ports in docker-compose.yml, so here is nothing to override. Besides, the goal of this file is not clear, it is better to point in its name that it is to be used for developing purposes

Comment (ebaranetsky) [reply products/docker-compose.override.yml:1]: It  will add additional usage compelexity, rigth now it is only :
`` docker-compose up``
If we rename this file the command above will look like:
``docker-compose  -f docker-compose.yml -f docker-compose-dev.yml up``
Let's keep it simple.

File: products/core/src/test/java/org/alliedium/ignite/migration/BadPathAsserter.java (hunk 3 above)
Comment (ebaranetsky) [reply products/core/src/test/java/org/alliedium/ignite/migration/BadPathAsserter.java]: we use junit in this project, yes there are standard ways but they are limited, there is no ability to verify exception message for example. Also we can use hamcrest library but checks of execption messages look huge and messy there

File: products/core/src/test/java/org/alliedium/ignite/migration/SupportedActionsTests.java (hunk 4 above)
Comment (ebaranetsky) [reply products/core/src/test/java/org/alliedium/ignite/migration/SupportedActionsTests.java]: It is outdated, but I will answer, we store a list of items to compare it to cache content after this cache will be serialized -> wiped out from ignite -> deserialized into ignite back.
The process of [serialize -> wipe -> deserialized -> check] is standard for most of tests at current repository state/code

File: products/core/src/main/java/org/alliedium/ignite/migration/Controller.java (hunk 2 above)
Comment (ebaranetsky) [reply products/core/src/main/java/org/alliedium/ignite/migration/Controller.java]: done, thank you

**Review by pgagarinov — APPROVED** [review]

**Pull request merged**